core/field_variable.js

```diff
@@ -84,14 +86,14 @@ Blockly.FieldVariable.prototype.initModel = function() {
     // Check if there was exactly one element specified in the
     // variableTypes list. This is the list that specifies which types of
     // variables to include in the dropdown.
-    // If there is exactly one element specified, make the variable
-    // being created this specified type. Else, default behavior is to create
-    // a scalar variable
-    if (this.getVariableTypes_().length == 1) {
-      this.sourceBlock_.workspace.createVariable(this.getValue(),
-        this.getVariableTypes_()[0]);
-    } else {
-      this.sourceBlock_.workspace.createVariable(this.getValue());
+    this.sourceBlock_.workspace.createVariable(this.getValue(), this.variableType);
+  } else {
+    // Using shorter name for this constant
+    var broadcastMsgType = Blockly.BROADCAST_MESSAGE_VARIABLE_TYPE;
+    var broadcastVars = this.sourceBlock_.workspace.getVariablesOfType(broadcastMsgType);
+    if (this.variableType == broadcastMsgType && broadcastVars.length != 0) {
+      broadcastVars.sort(Blockly.VariableModel.compareByName);
```

Comment:
Can you check that this isn't destroying anything (sort is in-place)

Reply:
Confirmed, it's not destroying anything. The result of `this.sourceBlock_.workspace.getVariablesOfType(broadcastMsgType)` does not change, but broadcastVars (which is specifically being used for the sort) does.

![image](https://user-images.githubusercontent.com/1786240/34311918-8ac18c82-e72e-11e7-9186-b48248a17727.png)